src/main/java/games/strategy/engine/data/GameParser.java

```diff
@@ -66,43 +66,69 @@
+  private GameParser(final String mapName) {
+    this.mapName = mapName;
   }
 
   /**
-   * Parses the XML Document provided by the given InputStream into a DOM.
-   * 
-   * @return The root Element of the DOM
+   * Performs a deep parse of the game definition contained in the specified stream.
+   *
+   * @return A complete {@link GameData} instance that can be used to play the game.
    */
-  private Element parseDom(final InputStream stream) throws SAXException {
-    Preconditions.checkNotNull(stream, "InputStream must be non-null!");
-    return getDocument(stream).getDocumentElement();
+  public static GameData parse(final String mapName, final InputStream stream)
+      throws GameParseException, SAXException, EngineVersionException {
+    return parse(mapName, stream, new AtomicReference<>());
   }
 
   /**
-   * Parses a file into a GameData object.
+   * Performs a deep parse of the game definition contained in the specified stream.
+   *
+   * @param gameName Receives the game name parsed from the game definition.
+   *
+   * @return A complete {@link GameData} instance that can be used to play the game.
    */
-  public GameData parse(final InputStream stream, final AtomicReference<String> gameName)
+  public static GameData parse(final String mapName, final InputStream stream, final AtomicReference<String> gameName)
+      throws GameParseException, SAXException, EngineVersionException {
+    checkNotNull(mapName);
+    checkNotNull(stream);
+    checkNotNull(gameName);
+
+    return new GameParser(mapName).parse(stream, gameName);
+  }
+
+  private GameData parse(final InputStream stream, final AtomicReference<String> gameName)
       throws GameParseException, SAXException, EngineVersionException {
     final Element root = parseDom(stream);
     parseMapProperties(root, gameName);
-    // everything until here is needed to select a game
     parseMapDetails(root);
     return data;
   }
 
   /**
-   * Parses just the essential parts of the maps.
-   * Used to display all available maps.
+   * Performs a shallow parse of the game definition contained in the specified stream.
+   *
+   * @return A partial {@link GameData} instance that can be used to display metadata about the game (e.g. when
+   *         displaying all available maps); it cannot be used to play the game.
    */
-  public GameData parseMapProperties(final InputStream stream, final AtomicReference<String> gameName)
+  public static GameData parseShallow(final String mapName, final InputStream stream)
+      throws GameParseException, SAXException, EngineVersionException {
+    checkNotNull(mapName);
+    checkNotNull(stream);
+
+    return new GameParser(mapName).parseShallow(stream);
+  }
+
+  private GameData parseShallow(final InputStream stream)
       throws GameParseException, SAXException, EngineVersionException {
     final Element root = parseDom(stream);
-    parseMapProperties(root, gameName);
+    parseMapProperties(root, new AtomicReference<>());
```

Comment:
Passing an empty game name does not make a lot of sense to me, @ssoloff should the API of parseMapProperties be updated to not accept a second parameter, or would a comment explain it a bit better?

Reply:
Sorry, missed this question...

I agree with the smell identification here, but `parseMapProperties()` is private, so this detail hasn't really leaked out through the public API.

Kind of related to your other comment, the `gameName` output parameter should indeed be replaced with something else.  My initial thought was to extract this information to `GameData` so it can be communicated though the `GameData` instance we're already returning from the `parseXXX` methods.  If that's not possible or doesn't make sense, then using a composite value for the return type that aggregates the `GameData` and the `String` game name would probably work, too.  That is:

```java
public class GameParser {
  public class Result {
    public final GameData gameData;
    public final String gameName;

    // ... elided ...
  }

  public static Result parse(final String mapName, final InputStream stream)
      throws GameParseException, SAXException, EngineVersionException {
    // ... elided ...
  }
}
```

Reply:
Actually, looking at this again, the `gameName` parameter is simply set to `GameData#getGameName()`, so this information is _already_ captured in `GameData`.  Sounds like removing the `gameName` parameter outright is the right thing to do.